### PR TITLE
Ensure that retrying published changesets resets failure message

### PR DESCRIPTION
**Pull request by mrnugget**
Discovered this while investigating #12700.

Edit: I wrote up more thoughts on this PR and other options in https://github.com/sourcegraph/sourcegraph/issues/12700#issuecomment-671798531.

### CODE REVIEW

File: enterprise/internal/campaigns/reconciler.go [hunk 1]
```diff
@@ -199,7 +199,8 @@ func (r *reconciler) updateChangeset(ctx context.Context, tx *Store, ch *campaig
 	// If we only need to update the diff, we're done, because we already
 	// pushed the commit. We don't need to update anything on the codehost.
 	if !delta.NeedCodeHostUpdate() {
-		return nil
+		ch.FailureMessage = nil
```
Comment (eseliger) [inline enterprise/internal/campaigns/reconciler.go:202]: Small optimization: only do that if `ch.FailureMessage != nil`

Comment (mrnugget) [reply enterprise/internal/campaigns/reconciler.go:202]: I think always setting it to nil might even be faster than adding the check :)

Comment (eseliger) [reply enterprise/internal/campaigns/reconciler.go:202]: In that case we wouldn't have needed the entire "save to db" roundtrip :) but doesn't matter too much I guess

Comment (mrnugget) [reply enterprise/internal/campaigns/reconciler.go:202]: Ah, yes, you're right, I missed that. But yeah, I don't think it matters that much for now and we have the nice effect of `updated_at` being updated ;)

**Pull request merged**